main.go

```diff
@@ -274,3 +287,13 @@ func isOpenShift(ctx context.Context, dc discovery.DiscoveryInterface) bool {
 	logger.Info("We detected being on Vanilla Kubernetes!")
 	return false
 }
+
+func getOpenShiftDomainName(ctx context.Context, routeClient routev1ClientSet.Interface) (string, error) {
+	route, err := routeClient.RouteV1().Routes(NamespaceConsoleLink).Get(ctx, NameConsoleLink, metav1.GetOptions{})
```

Comment:
Should it be more sourced from the `config.openshift.io/Ingress` resource named `cluster`?